source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -318,11 +315,23 @@ Http::FilterHeadersStatus OAuth2Filter::decodeHeaders(Http::RequestHeaderMap& he
     const std::string escaped_redirect_uri =
         Http::Utility::PercentEncoding::encode(redirect_uri, ":/=&?");
 
-    const std::string new_url = fmt::format(
-        AuthorizationEndpointFormat, config_->authorizationEndpoint(), config_->clientId(),
-        config_->encodedAuthScopes(), escaped_redirect_uri, escaped_state);
-
-    response_headers->setLocation(new_url + config_->encodedResourceQueryParams());
+    Http::Utility::Url authorization_endpoint;
+    authorization_endpoint.initialize(config_->authorizationEndpoint(),
```

Comment:
Added a check in the constructor to make sure the authorization endpoint is a valid URL (consistent with how oauth_token_endpoint_.cluster() is checked) and an ASSERT here. Seems preferable to returning sendUnauthorizedResponse() as an error is returned when envoy is started rather than on the first oauth request, but please let me know if you have any concerns.